controllers/dataprotection/backup_controller.go

```diff
@@ -638,6 +644,26 @@ func (r *BackupReconciler) deleteExternalResources(
 	return r.deleteExternalJobs(reqCtx, backup)
 }
 
+func (r *BackupReconciler) updateBackupExpiredTime(
```

Comment:
`updateExpirationByCompletionTime` may be a more explicit function name.

And make sure test case cover it.

Reply:
the last stage of `handleRunningPhase` may be a better position to update its expiration.